sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -823,24 +823,34 @@ public Mono<MsalToken> authenticateWithBrowserInteraction(TokenRequestContext re
             return Mono.error(LOGGER.logExceptionAsError(new RuntimeException(e)));
         }
 
-        if (options.isBrokerEnabled() && options.useOperatingSystemAccount()) {
-            return getPublicClientInstance(request).getValue().flatMap(pc ->
-                Mono.fromFuture(() ->
-                    acquireTokenFromPublicClientSilently(request, pc, null, false)).
-                    map(MsalToken::new));
-        } else {
-
+        // If the broker is enabled, try to get the token for the default account by passing
+        // a null account to MSAL. If that fails, show the dialog.
+
+        return getPublicClientInstance(request).getValue().flatMap(pc -> {
+            if (options.isBrokerEnabled() && options.useDefaultBrokerAccount()) {
+                return Mono.fromFuture(() ->
+                    acquireTokenFromPublicClientSilently(request, pc, null, false))
+                    .onErrorResume(e -> Mono.empty());
```

Comment:
May be a little pedantic, would it be useful to have the exception being resumed here be maintained? The code would have to duplicate a bit where there wouldn't be a `switchIfEmpty` and that code block duplicate (mostly),

```java
        if (options.isBrokerEnabled() && options.useOperatingSystemAccount()) {
            return getPublicClientInstance(request).getValue()
                .flatMap(pc -> Mono.fromFuture(() -> acquireTokenFromPublicClientSilently(request, pc, null, false))
                    .onErrorResume(e -> Mono.defer(() -> {
                        Mono<IAuthenticationResult> acquireToken = Mono.fromFuture(() -> pc.acquireToken(
                            buildInteractiveRequestParameters(request, loginHint, redirectUri).build()));
    
                        return acquireToken.onErrorMap(t -> {
                            t.addSuppressed(e);
                            return new ClientAuthenticationException(
                                "Failed to acquire token with Interactive Browser Authentication.", null, t);
                        });
                    })))
                .map(MsalToken::new);
        } else {

            InteractiveRequestParameters.InteractiveRequestParametersBuilder builder =
                buildInteractiveRequestParameters(request, loginHint, redirectUri);

            SynchronizedAccessor<PublicClientApplication> publicClient = getPublicClientInstance(request);

            Mono<IAuthenticationResult> acquireToken = publicClient.getValue()
                .flatMap(pc -> Mono.fromFuture(() -> pc.acquireToken(builder.build())));

            return acquireToken.onErrorMap(t -> new ClientAuthenticationException(
                "Failed to acquire token with Interactive Browser Authentication.", null, t)).map(MsalToken::new);
        }
```

Right now, the error resumed there is lost, which may contain some useful information on fixing a problem with brokering.

Reply:
The error message is being ignored right now because it is really control flow. MSAL will either return a valid token or throw an exception. This operation is speculative; for various reasons automatic sign-in might fail and we fall back to showing the UI. Preserving the error here doesn't provide any useful log context, nor does it provide anything actionable down the line.

If you feel strongly here I'll make a change, but I don't see the upside. I will put a comment in explaining why we're ignoring it, though.

Reply:
Makes sense, didn't really know that, was more worried about losing exception information if the fallback failed as well but this isn't really a case where the exception being ignored is meaningful.

Given all that, mind leaving a small comment about why it's okay for the exception to be lost in this scenario.